blocks/market-insight/market-insight.js

```diff
@@ -0,0 +1,209 @@
+    // Social share button
+    const socialShare = createDiv('div', 'socialshare');
+    const button = createDiv('button', '');
+    const image = createDiv('img', '');
+    image.src = '../../icons/gray-share-icon.svg';
+    image.alt = 'gray-share-icon';
+    button.append(image);
+    addSocialButtonEvent(button, block);
+    socialShare.append(button);
+    powerby.append(socialShare);
+    li.append(title);
+    li.append(description);
+    li.append(powerby);
+    ul.append(li);
+  }
+  return ul;
+}
+
+function addModalCloseEvent(closeItem, modal) {
+  closeItem.addEventListener('click', () => {
+    modal.classList.toggle('visible');
+    const body = document.querySelector('body');
+    body.classList.toggle('modal-open');
+  });
+}
+
+function addModalOuterCloseEvent(modal) {
+  modal.addEventListener('click', (e) => {
+    if (modal.classList.contains('visible') && e.target === modal) {
+      modal.classList.toggle('visible');
+      const body = document.querySelector('body');
+      body.classList.toggle('modal-open');
+    }
+  });
+}
+
+function createSocialIcons(modalBody) {
+  const div = createDiv('div', '');
+  const ul = createDiv('ul', '');
+  const socialList = ['whatsapp', 'facebook', 'linkedin', 'twitter', 'copy-link'];
+  [...socialList].forEach((item) => {
+    const li = createDiv('li', '');
+    const link = createDiv('a', '');
+    link.target = '_blank';
+    link.classList.add(item);
+    const img = createDiv('img', '');
+    img.src = `../../icons/${item}-icon.png`;
+    img.alt = `${item}`;
+    link.append(img);
+    li.append(link);
+    ul.append(li);
+  });
+  div.append(ul);
+  modalBody.append(div);
+}
+
+function decorateModal(blockCfg) {
+  const modalTitleContent = (blockCfg['modal-title'] ?? '');
+  const modal = createDiv('div', 'modal');
+  const modalDislog = createDiv('div', 'modal-dislog');
+  const modalContent = createDiv('div', 'modal-content');
+  const modalBody = createDiv('div', 'modal-body');
+  const closeButton = createDiv('button', 'close-button');
+  const closeIcon = createDiv('span', '');
+  closeIcon.innerHTML = '&times;';
+  closeButton.append(closeIcon);
+  addModalCloseEvent(closeButton, modal);
+  const modalTitle = createDiv('div', '');
+  const h3 = createDiv('h3', '');
+  const strongTag = createDiv('strong', '');
+  strongTag.append(modalTitleContent);
+  h3.append(strongTag);
+  modalTitle.append(h3);
+  modalBody.append(modalTitle);
+  createSocialIcons(modalBody);
+  modalContent.append(modalBody);
+  modalContent.append(closeButton);
+  modalDislog.append(modalContent);
+  modal.append(modalDislog);
+  addModalOuterCloseEvent(modal);
+
+  return modal;
+}
+
+export default async function decorate(block) {
+  const placeholders = await fetchPlaceholders();
+  const blockCfg = readBlockConfig(block);
+  const title = decorateTitle(blockCfg);
+  const discoverMoreButton = decorateDiscoverMore(blockCfg, placeholders);
+  const results = await fetchMarketInsightMockData();
```

Comment:
loading dynamic content should be done with observer here https://github.com/aemsites/icicidirect/blob/main/scripts/blocks-utils.js#L83

Reply:
Nice, made the method to support multiple arguments.